include/grpc/grpc_security.h

```diff
@@ -808,49 +808,81 @@ GRPCAPI int grpc_tls_key_materials_config_set_key_materials(
     grpc_tls_key_materials_config* config);
 
 /** --- TLS credential reload config. ---
-    It is used for experimental purpose for now and subject to change.*/
+    Note the API is experimental now and subject to change.*/
 
 typedef struct grpc_tls_credential_reload_arg grpc_tls_credential_reload_arg;
 
 /** A callback function provided by gRPC to handle the result of credential
     reload. It is used when schedule API is implemented asynchronously and
-    serves to bring the control back to grpc C core. It is used for
-    experimental purpose for now and subject to change. */
+    serves to bring the control back to grpc C core.
+    Note the API is experimental now and subject to change. */
 typedef void (*grpc_tls_on_credential_reload_done_cb)(
     grpc_tls_credential_reload_arg* arg);
 
 /** A struct containing all information necessary to schedule/cancel a
     credential reload request.
-    - cb and cb_user_data represent a gRPC-provided
-      callback and an argument passed to it.
-    - key_materials_config is an in/output parameter containing currently
-      used/newly reloaded credentials. If credential reload does not result in
-      a new credential, key_materials_config should not be modified. The same
-      key_materials_config object can be updated if new key materials is
-      available.
-    - status and error_details are used to hold information about
-      errors occurred when a credential reload request is scheduled/cancelled.
-    - config is a pointer to the unique grpc_tls_credential_reload_config
+    Note that we currently only support synchronous credential reloading. All
+    the fields related to asynchronous credential reloading can be ignored for
+    now.
+    - cb: a callback-function provided by gRPC stack for users to indicate an
+      asynchronous reloading operation is complete. Users are expected to call
+      this function when asynchronous reloading is done.
+      This field is only applicable in the asynchronous mode which is currently
+      not implemented.
+    - cb_user_data: a pointer used to retrieve some implementations in C core.
+      This is needed because after users call |cb|, we will use this field to
+      notify the gRPC stack to continue the handshake.
+      Users are not expected to interact with this field directly.
+    - key_materials_config: an in/output parameter containing currently
+      used/newly reloaded credentials. This field is provided by the gRPC stack,
+      and users can use it to handle credential reloading.
+      Note that, key_materials_config is not a per-connection parameter, which
+      means it might be shared by multiple connections.
+    - status: a per-connection parameter to indicate the finish status of a
+      schedule/cancel request. Users are expected to manage this field properly
+      in order to do credential reloading.
+      If set to |GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_NEW|, gRPC stack would
```

Comment:
I think it's a good idea to explain the relation between `key_materials_config` and `status` here. I would suggest that those two fields should be consistent in a sense that if `key_materials_config` is updated, then status must be `GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_NEW`, and no `err_details` should be specified, so on.